src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java

```diff
@@ -188,7 +189,8 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .requestMatchers("/management/prometheus/**").access((authentication, context) -> new AuthorizationDecision(monitoringIpAddresses.contains(context.getRequest().getRemoteAddr())))
                 .requestMatchers("/**").authenticated()
             )
-            .with(securityConfigurerAdapter(), configurer -> configurer.configure(http));
+            .with(securityConfigurerAdapter(), configurer -> configurer.configure(http))
+            .httpBasic(Customizer.withDefaults());
```

Comment:
To me it would not be obvious that this line adds http basic authentication. I suggest we better document this class and ideally add natural language comment with a justification for the configuration in each line so that people who are not familiar with Spring security can easily understand it.